internal/pkg/service/common/etcdop/op/txn_test.go

```diff
@@ -412,8 +412,12 @@ func TestTxnOp_And_RealExample(t *testing.T) {
 	txn.And(deleteOp)
 	txn.Then(etcdop.Key("key/txn/succeeded").Put(client, "true"))
 	txn.Else(etcdop.Key("key/txn/succeeded").Put(client, "false"))
-	txn.OnResult(func(r *TxnResult[NoResult]) {
-		_, _ = fmt.Fprintf(&log, "txn succeeded: %t\n", r.Succeeded())
+	txn.AddProcessor(func(ctx context.Context, r *TxnResult[NoResult]) {
+		if err := r.Err(); err != nil {
+			_, _ = fmt.Fprintf(&log, "txn succeeded: error: %s", strings.ReplaceAll(err.Error(), "\n", ";"))
+		} else {
+			_, _ = fmt.Fprintf(&log, "txn succeeded: %t\n", r.Succeeded())
+		}
 	})
```

Comment:
Updated tests:

- `OnResult` callback is invoked if there is no error.
- Error can be generated by a previous callback.
- Therefore, because the top-level callbacks have been moved to the end, now they may be called after some other callback that generate errors.
- Result was that the `txn succeeded: ` message disappeared from some tests.

So I replaced `OnResult` shortcut with `AddProcessor` method - the callback is called always.
- I also distinguish between an error and a failed transaction (some IF condition was not met).
- See changes bellow, it should be clear.